forge.config.js

```diff
@@ -40,6 +40,7 @@ const deletePycFiles = async (directory) => {
 /** @type {import("@electron-forge/shared-types").ForgeConfig} */
 const config = {
     packagerConfig: {
+        asar: true,
```

Comment:
What's this?

Because of this option, there is a 200MB `app.asar` in all windows builds now.

Reply:
Okay, `asar: true` itself isn't the problem, it just hides it. Setting `asar: false` and making a build reveals that vite copies the ***entire `node_modules` folder***.

![image](https://github.com/chaiNNer-org/chaiNNer/assets/20878432/eb49eac8-da7d-446e-9ad2-2c609bf860eb)

The issue seems to be that vite doesn't inline our dependencies in `main/main.ts`

Reply:
Something somewhere told me to enable it. Hover over it in vscode and itll explain what it is

Reply:
I looked it up. It's necessary for macos, so we have to do asar. But as I said, asar is not the problem. vite is.

Reply:
Is this what you want it to do? https://www.npmjs.com/package/vite-plugin-singlefile

Reply:
I can't find any other information on inlining node_modules. Are you sure that's something we want to do in the first place?

Reply:
No. Firstly, this isn't about the renderer, so a single HTML file wouldn't be useful. Secondly, node_module in itself isn't a problem. 

The problem is that vite includes **all** our dependencies. It doesn't so any sort of tree shaking (or similar) to only include the dependencies we need. Almost everything inside the `node_modules` folder is dead weight. I mean, I only noticed this because our .zip distribution went from 100MB to 150MB. Same for the installer.

Reply:
I see. That definitely seems wrong, we're probably missing something like what Babel had with the exclude, maybe for the esbuild and/or SWC settings. Should be an easy fix if we just need to ignore it

Reply:
Apparently we can use https://www.npmjs.com/package/rollup-plugin-node-externals for it. I tried this, but it makes us unable to import the wasm file for rregex 🙃

Reply:
That's not ideal.

Reply:
forgot to give some updates. i've tried several things but just keep running into further issues. will take some more time to investigate why and resolve (if even possible). this is the kind of issue that feels like it should be really simple to fix and yet seems basically impossible, which is really starting to piss me off and makes me want to just give up on this.

Reply:
One idea: If you got everything except for the wasm file working, maybe we could inline the wasm file?

I'm thinking about using a plugin to replace our `import wasmUrl from "rregexp/lib/rregex.wasm?url"` with a data URL along the lines of `const wasmUrl = "data:blob;base64, ...";` Since we're inlining the wasm, it can't be missing, and since we're doing it ourselves, vite can't fuck it up.

Reply:
The problem isn't the wasm file anymore. I handled that. It's just the entire process of trying to get it to externalize the dependencies. I got it working to a certain point, but I got stuck when electron-log was complaining about being initialized twice, somehow

Reply:
I think it's safe to say this has defeated me yet again. I cannot for the life of me figure out how to do this properly